drift.py

```diff
@@ -1,146 +1,90 @@
-            result.commit = commit.hash
-            result.abstract = name
-            result.timestamp = str(commit.author_date)
-            if len(spec[verToken:]) > 0:
-                result.version = spec[verToken+1:]
-            if out.returncode == SUCCESS:
-                concrete_spec = json.loads(out.stdout)["spec"]
-                # In the case that the concretized spec hash changes
-                # we want to record the commit where this happens.
-                if self.last[spec] != "" and self.last[spec][0][name]["full_hash"] != concrete_spec[0][name]["full_hash"]:
-                    # Check if the package version is different.
-                    if self.last[spec][0][name]["version"] != concrete_spec[0][name]["version"]:
-                        result.tags += ["pkg-updated"]
-                    # +---------------------+
-                    # |     Dependencies    |
-                    # +---------------------+
-                    result.tags += set(self.tag_deps(concrete_spec, spec, name))
-                    # +---------------------+
-                    # |      Variants       |
-                    # +---------------------+
-                    result.tags += set(self.tag_variants(concrete_spec, spec, name))
-                    # Send Inflection Point
-                    send(result)
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            name = abstract_spec
+            version = ""
+            if "@" in abstract_spec:
+                name, version = abstract_spec.split("@", 1)
+            # Check that version exists in package.py file.
+            out = run(f"spack versions --safe-only {abstract_spec}")
+            # Don't attempt to concretize if the version doesn't yet exist.
+            if version in out.stdout:
+                out = run(f"spack spec --yaml {abstract_spec}")
+                # If concretization successful check the resulting concrete specs.
+                if out.returncode == SUCCESS:
+                    concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last:
+                        self.last[abstract_spec] = concrete_spec
+                    if self.last[abstract_spec] == concrete_spec:
+                        continue
+                    # If the spec is different check what is different and save as tags.
+                    diff = compare_specs(
+                        self.last[abstract_spec],
+                        concrete_spec,
+                        colorful=False)
```

Comment:
Oh man, we need to make this consistent with other `color=` kwargs.  Guess this is fine for now.

Reply:
oops sorry that's my bad

Reply:
Fixed in https://github.com/spack/spack/pull/25292